src/lmdb.ml

```diff
@@ -44,6 +46,12 @@ let () =
 let pp_error fmt i =
   Format.fprintf fmt "%s@." (mdb_strerror i)
 
+type 'a cap =
+  | Ro : [ `Read ] cap
+  | Rw : [ `Read | `Write ] cap
+let ro = Ro
+let rw = Rw
+
```

Comment:
> Replace all the [`Read] by _, make things covariant

I'm not sure how your proposal would work out with stricter capabilities typing on transactions, cursors, environments and ``S.create``.
Explicitly writing out ``['Read]``, ``['Write]``, and ``['Read | 'Write] capabilities may seem overengineered, but it is clearer.
With covariance you could ask for a broken read/write transaction:
``` ocaml
Txn.go (ro :> [ `Write ]) env (f : [ `Write ] txn -> _ option)
```
I would ask you to please accept this scheme for now and see how it will unfold in the richer typed interface. If it in the end is unnecessary we can still revert the change.
I do not want to expose the GADT in the interface (yet), to be able to change the internal representation if they turn out to complicate the typing too much (I'm not experienced with locally abstract types...).

Reply:
Alright, let's keep moving, and re-assess when the rest of the API stabilizes.

> I'm not experienced with locally abstract types

Fortunately, I am :p